src/Arcus.Messaging.Pumps.Abstractions/MessageHandling/MessageHandler.cs

```diff
@@ -117,7 +117,7 @@ public Type GetMessageHandlerType()
         /// </summary>
         /// <typeparam name="TMessageContext">The type of the message context.</typeparam>
         /// <param name="messageContext">The context in which the incoming message is processed.</param>
-        public bool CanProcessMessage<TMessageContext>(TMessageContext messageContext) where TMessageContext : MessageContext
+        public bool CanProcessMessageBasedOnContext<TMessageContext>(TMessageContext messageContext) where TMessageContext : MessageContext
```

Comment:
This is abreaking change, can we create a new method and deprecate the old one?

Reply:
Well, I don't know if we publicly support this type. It's for internal use only and is only public bc we can use it in our tests and Azure Functions project easily. Not sure if we should do this, but we can of course.

Reply:
Well if it's public then it's supported🤷‍♂️

We don't know if somebody is extending it so have to support it I'm afraid

Reply:
Voila, done. Resolving.